glean-core/android/src/main/java/mozilla/telemetry/glean/net/HttpURLConnectionUploader.kt

```diff
@@ -104,9 +106,16 @@ class HttpURLConnectionUploader : PingUploader {
     }
 
     @Throws(IOException::class)
-    internal fun doUpload(connection: HttpURLConnection, data: String): Int {
-        connection.outputStream.bufferedWriter().use {
-            it.write(data)
+    internal fun doUpload(connection: HttpURLConnection, data: ByteArray, isGzipped: Boolean): Int {
+        connection.outputStream.use {
+            if (isGzipped) {
```

Comment:
There's no need for this. Even if it is plaintext we should be able to write the ByteArray as is. Therefore not requiring the `isGzipped` flag at all.

Reply:
I... wow, you're right.